flake.nix

```diff
@@ -0,0 +1,62 @@
+{
+  description = "Install Asustor kernel modules on NixOS";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  };
+
+  outputs = { self, nixpkgs }:
+    let
+      system = "x86_64-linux";
+      pkgs = nixpkgs.legacyPackages.${system};
+      module =
+        ({ stdenv, lib, kernel, kmod, ... }:
+
+          stdenv.mkDerivation rec {
+            name = "asustor-platform-driver-${version}-${kernel.version}";
+            version = "0.1";
+
+            src = ./.;
+
+            hardeningDisable = [ "pic" "format" ];
+            # Install kmod for depmod dependency if we need it
+            #nativeBuildInputs = kernel.moduleBuildDependencies ++ [ kmod ];
```

Comment:
What's the motivation for leaving this commented? If it's a common-use case, is there a better way to support it?

Reply:
I noticed that `depmod` is used in the `Makefile`, but everything seems to work without. Is it OK, if I skip it?

Reply:
`depmod` is required for `modprobe` to work properly, I think it uses it to figure out what other modules to implicitly load, so it should be run after new modules have been installed.
It might have worked on your system without depmod by chance, but I wouldn't rely on that behavior.

Reply:
I will remove it, because it seems to work and I saw that line being disabled in other NixOS kernel modules. It probably works differently.

Reply:
I'm not familiar with Nix, but this is surprising.
You may wanna ask other Nix developers/maintainers who may know about such things why it's not needed there (maybe Nix calls depmod or something equivalent automatically for kernel modules? Just a possible explanation, I really don't know)

Reply:
I asked and they said it is not required during build time.